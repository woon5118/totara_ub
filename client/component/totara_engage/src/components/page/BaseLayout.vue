<!--
  This file is part of Totara Enterprise Extensions.

  Copyright (C) 2020 onwards Totara Learning Solutions LTD

  Totara Enterprise Extensions is provided only to Totara
  Learning Solutions LTD's customers and partners, pursuant to
  the terms and conditions of a separate agreement with Totara
  Learning Solutions LTD or its affiliate.

  If you do not have an agreement with Totara Learning Solutions
  LTD, you may not access, use, modify, or distribute this software.
  Please contact [licensing@totaralearning.com] for more information.

  @author Kian Nguyen <kian.nguyen@totaralearning.com>
  @module totara_engage
-->

<template>
  <Grid :max-units="String(maxPageUnits)">
    <GridItem :units="parseInt(contentUnits)">
      <slot name="content" />
    </GridItem>

    <GridItem :units="sidePanelUnits">
      <slot name="sidepanel" />
    </GridItem>
  </Grid>
</template>

<script>
import Grid from 'tui/components/grid/Grid';
import GridItem from 'tui/components/grid/GridItem';

export default {
  components: {
    Grid,
    GridItem,
  },

  props: {
    maxPageUnits: {
      type: [Number, String],
    },

    contentUnits: {
      type: [Number, String],
      required: true,
    },

    sidePanelUnits: {
      type: [Number, String],
      required: true,
    },
  },
};
</script>
