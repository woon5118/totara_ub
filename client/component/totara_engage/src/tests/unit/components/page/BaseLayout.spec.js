/**
 * This file is part of Totara Enterprise Extensions.
 *
 * Copyright (C) 2020 onwards Totara Learning Solutions LTD
 *
 * Totara Enterprise Extensions is provided only to Totara
 * Learning Solutions LTD's customers and partners, pursuant to
 * the terms and conditions of a separate agreement with Totara
 * Learning Solutions LTD or its affiliate.
 *
 * If you do not have an agreement with Totara Learning Solutions
 * LTD, you may not access, use, modify, or distribute this software.
 * Please contact [licensing@totaralearning.com] for more information.
 *
 * @author Kian Nguyen <kian.nguyen@totaralearning.com>
 * @module totara_engage
 */

import BaseLayout from 'totara_engage/components/page/BaseLayout';
import { shallowMount } from '@vue/test-utils';

describe('totara_engage/components/page/BaseLayout', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = shallowMount(BaseLayout, {
      propsData: {
        contentUnits: 7,
        sidePanelUnits: 5,
      },
    });
  });

  it('Checks snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
