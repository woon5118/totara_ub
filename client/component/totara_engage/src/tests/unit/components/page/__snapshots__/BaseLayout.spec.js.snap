// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`totara_engage/components/page/BaseLayout Checks snapshot 1`] = `
<grid-stub
  direction="horizontal"
  gridtag="div"
  maxunits="undefined"
  usehorizontalgap="true"
  useverticalgap="true"
>
  <griditem-stub
    griditemtag="div"
    hyphens="true"
    shrinks="true"
    sizedata="[object Object]"
    units="7"
  />
   
  <griditem-stub
    griditemtag="div"
    hyphens="true"
    shrinks="true"
    sizedata="[object Object]"
    units="5"
  />
</grid-stub>
`;
